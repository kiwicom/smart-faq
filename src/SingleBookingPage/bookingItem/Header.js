// @flow

import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import { Typography } from '@kiwicom/orbit-components';

import bookingStatus from '../../common/booking/bookingStatuses';
import formatBookingId from '../../helpers/formatBookingId';
import type { Header_booking } from './__generated__/Header_booking.graphql';

type Props = {|
  booking: Header_booking,
|};

const Header = ({ booking }: Props) => {
  const status = booking.status && bookingStatus[booking.status];

  return (
    <div>
      <div className="headerAbove">
        <div>
          {booking.databaseId && (
            <Typography type="secondary">
              Upcoming trip # {formatBookingId(booking.databaseId)}
            </Typography>
          )}
        </div>
        <div>
          <Typography size="small" type="active">
            select other booking
          </Typography>
        </div>
      </div>
      <div className="headerTitle">
        <Typography size="header" type="attention">
          Upcoming booking
        </Typography>
      </div>
      <div className="headerBelow">
        {status && (
          <Typography>
            <span style={{ color: status.color }}>{status.text}</span>
          </Typography>
        )}
      </div>
      <style jsx>
        {`
          .headerAbove,
          .headerTitle {
            margin-bottom: 4px;
          }
          .headerAbove {
            display: flex;
            justify-content: space-between;
          }
          .headerBelow {
            margin-bottom: 16px;
          }
        `}
      </style>
    </div>
  );
};

export default createFragmentContainer(
  Header,
  graphql`
    fragment Header_booking on BookingInterface {
      databaseId
      status
    }
  `,
);
