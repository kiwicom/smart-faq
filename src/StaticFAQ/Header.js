// @flow

import * as React from 'react';
import css from 'styled-jsx/css';
import { Link } from 'react-router-dom';
import { Typography } from '@kiwicom/orbit-components';

import routeDefinitions from '../routeDefinitions';
import CloseButton from '../common/CloseButton';
import BackButton from '../common/BackButton';

const style = css`
  .static-faq-header {
    width: 100%;
    height: 64px;
    box-shadow: inset 0 -1px 0 0 #e8edf1;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .leftButton {
    position: absolute;
    top: 23px;
    left: 16px;
  }
`;

type Props = {|
  leftButton: 'SignIn' | 'Back',
|};

const SignInButton = () => (
  <Link to={routeDefinitions.SIGN_IN} style={{ textDecoration: 'none' }}>
    <Typography type="attention" variant="bold">
      Sign In
    </Typography>
  </Link>
);

const Header = (props: Props) => (
  <React.Fragment>
    <CloseButton />
    {props.leftButton === 'SignIn' ? (
      <div className="leftButton">
        <SignInButton />
      </div>
    ) : (
      <BackButton text="Back" link={routeDefinitions.STATIC_FAQ} />
    )}
    <div className="static-faq-header">
      <Typography size="header" type="attention" variant="bold">
        Help
      </Typography>
    </div>
    <style jsx>{style}</style>
  </React.Fragment>
);

export default Header;
