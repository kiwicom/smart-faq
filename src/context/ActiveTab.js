// @flow
/* eslint-disable react/no-unused-state */

import * as React from 'react';

export type TabType = 'search' | 'summary' | 'account' | null;
type Props = {|
  children: React.Node,
|};
type StateValues = {
  activeTab: TabType,
};

type StateCallbacks = {
  activateTab: (tab: TabType) => void,
  toggleTab: (tab: TabType) => void,
};

export type ActiveTabType = StateValues & StateCallbacks;

const initialState: StateValues = {
  activeTab: null,
};

export const ActiveTab = React.createContext({
  ...initialState,
  activateTab: (tab: TabType) => null, // eslint-disable-line no-unused-vars
  toggleTab: (tab: TabType) => null, // eslint-disable-line no-unused-vars
});

class ActiveTabProvider extends React.Component<Props, ActiveTabType> {
  constructor(props: Props) {
    super(props);

    const activateTab = (tab: TabType) => {
      this.setState({ activeTab: tab });
    };

    const toggleTab = (tab: TabType) => {
      this.setState(prevState => ({
        activeTab: prevState.activeTab === tab ? null : tab,
      }));
      this.forceUpdate();
    };

    this.state = {
      ...initialState,
      activateTab: activateTab,
      toggleTab: toggleTab,
    };
  }

  render() {
    return (
      <ActiveTab.Provider value={this.state}>
        {this.props.children}
      </ActiveTab.Provider>
    );
  }
}

export default ActiveTabProvider;
