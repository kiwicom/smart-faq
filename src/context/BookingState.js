// @flow
/* eslint-disable react/no-unused-state */

import * as React from 'react';

import type { onLogout } from '../types';

export const getFAQSection = ({
  hasBooking,
  isPastBooking,
  isUrgent,
}: BookingStateDescription) => {
  if (!hasBooking) return 'BEFORE_BOOKING';
  if (isPastBooking) return 'PAST_BOOKING';
  if (isUrgent) return 'URGENT_BOOKING';

  return 'UPCOMING_BOOKING';
};

const initialState = {
  FAQSection: null,
  bookingPage: 'SINGLE_BOOKING',
  selectedBooking: null,
};

export type FAQSectionType =
  | 'BEFORE_BOOKING'
  | 'UPCOMING_BOOKING'
  | 'URGENT_BOOKING'
  | 'PAST_BOOKING';

type Props = {
  children: React.Node,
  onLogout: onLogout,
  hasBooking: boolean,
  isPastBooking?: boolean,
  isUrgent?: boolean,
};

type StateValues = {
  FAQSection: ?FAQSectionType,
  bookingPage: 'SINGLE_BOOKING' | 'ALL_BOOKINGS',
  selectedBooking: ?number,
};

type StateCallbacks = {
  closeAllBooking: () => void,
  onDisplayAll: () => void,
  onSelectBooking: (id: number) => void,
  onSetFAQSection: (isUrgent: boolean, isPastBooking: boolean) => void,
  onLogout: onLogout,
};

type BookingStateDescription = {
  hasBooking: boolean,
  isPastBooking?: boolean,
  isUrgent?: boolean,
};

export type BookingStateType = StateValues & StateCallbacks;

export const BookingState = React.createContext({
  ...initialState,
  closeAllBooking: () => {},
  onDisplayAll: () => {},
  onSelectBooking: (id: number) => {}, // eslint-disable-line no-unused-vars
  onSetFAQSection: (isUrgent: boolean, isPastBooking: boolean) => {}, // eslint-disable-line no-unused-vars
  onLogout: () => Promise.resolve(true),
});

class BookingStateProvider extends React.Component<Props, BookingStateType> {
  constructor(props: Props) {
    super(props);
    const { hasBooking, isUrgent, isPastBooking } = props;

    this.state = {
      ...initialState,
      closeAllBooking: this.closeAllBooking,
      onLogout: this.onLogout,
      onSelectBooking: this.onClickSelect,
      onDisplayAll: this.onClickAllBooking,
      onSetFAQSection: this.onSetFAQSection,
      FAQSection: getFAQSection({ hasBooking, isUrgent, isPastBooking }),
    };
  }

  closeAllBooking = () => {
    this.setState({ bookingPage: 'SINGLE_BOOKING' });
  };

  onLogout = async () => {
    await this.props.onLogout();
    this.setState({ FAQSection: 'BEFORE_BOOKING' });
  };

  onClickAllBooking = () => {
    this.setState({ bookingPage: 'ALL_BOOKINGS' });
  };

  onClickSelect = (id: number) => {
    this.setState({ bookingPage: 'SINGLE_BOOKING', selectedBooking: id });
    const Body = document.querySelector('#SmartFAQ_Body');
    if (!Body) return;
    Body.dispatchEvent(new Event('scroll'));
  };

  onSetFAQSection = (isUrgent: boolean, isPastBooking: boolean) => {
    const section = getFAQSection({
      isUrgent,
      isPastBooking,
      hasBooking: true,
    });

    this.setState(({ FAQSection }) => {
      if (FAQSection !== section) {
        return { FAQSection: section };
      }
    });
  };

  render() {
    return (
      <BookingState.Provider value={this.state}>
        {this.props.children}
      </BookingState.Provider>
    );
  }
}

export const withLogout = <Props>(
  Component: React.ComponentType<{ onLogout: onLogout } & Props>,
) =>
  function withLogoutHOC(props: Props) {
    return (
      <BookingState.Consumer>
        {({ onLogout }: BookingStateType) => (
          <Component {...props} onLogout={onLogout} />
        )}
      </BookingState.Consumer>
    );
  };

export default BookingStateProvider;
