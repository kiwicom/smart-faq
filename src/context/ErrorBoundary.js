// @flow

import * as React from 'react';

const initialState = {
  error: false,
  errorDetails: {},
};

type Props = {
  children: React.Node,
};

export type State = {
  error: boolean,
  errorDetails: {},
};

export const ErrorBoundary = React.createContext({
  ...initialState,
});

class ErrorBoundaryProvider extends React.Component<Props, State> {
  state = initialState;
  componentDidCatch(error: Error, errorDetails: {}) {
    this.setState({
      error: true,
      errorDetails,
    });
  }

  render() {
    const { error, errorDetails } = this.state;

    return (
      <ErrorBoundary.Provider
        value={{
          error,
          errorDetails,
        }}
      >
        {this.props.children}
      </ErrorBoundary.Provider>
    );
  }
}

export default ErrorBoundaryProvider;
