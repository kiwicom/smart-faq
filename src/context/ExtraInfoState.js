// @flow

import * as React from 'react';

const initialState = {
  activeExtraInfoCategory: null,
};

export type ExtraInfoCategory = 'baggage' | 'boarding-passes';

type Props = {
  children: React.Node,
};

export type State = {
  activeExtraInfoCategory: ?ExtraInfoCategory,
};

export type ExtraInfoStateType = State & {
  toggleExtraInfoCategory: (category: ExtraInfoCategory) => void,
};

export const ExtraInfoState = React.createContext({
  ...initialState,
  toggleExtraInfoCategory: () => {}, // eslint-disable-line no-unused-vars
});

class ExtraInfoStateProvider extends React.Component<
  Props,
  ExtraInfoStateType,
> {
  constructor(props: Props) {
    super(props);

    this.state = {
      ...initialState,
      toggleBaggage: this.toggleBaggage, // eslint-disable-line react/no-unused-state
    };
  }

  toggleExtraInfoCategory = (category: ExtraInfoCategory) => {
    this.setState(prevState => ({
      activeExtraInfoCategory:
        prevState.activeExtraInfoCategory === category ? null : category,
    }));
  };

  render() {
    const { activeExtraInfoCategory } = this.state;

    return (
      <ExtraInfoState.Provider
        value={{
          activeExtraInfoCategory,
          toggleExtraInfoCategory: this.toggleExtraInfoCategory,
        }}
      >
        {this.props.children}
      </ExtraInfoState.Provider>
    );
  }
}

export default ExtraInfoStateProvider;
