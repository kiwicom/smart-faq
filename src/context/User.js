// @flow

import * as React from 'react';

import type { User, onLogin, onSocialLogin } from '../types';

export type UserContextType = {
  user: User,
  loginToken: ?string,
  onLogin: onLogin,
  onSocialLogin: onSocialLogin,
};

export const UserContext = React.createContext(
  ({
    user: null,
    loginToken: null,
    onLogin: (email, password) => Promise.resolve({ email, password }),
    onSocialLogin: provider => Promise.resolve(provider),
  }: UserContextType),
);

export const withLogin = <Props>(
  Component: React.ComponentType<{ onLogin: onLogin } & Props>,
) =>
  function withLoginHOC(props: Props) {
    return (
      <UserContext.Consumer>
        {({ onLogin }: UserContextType) => (
          <Component {...props} onLogin={onLogin} />
        )}
      </UserContext.Consumer>
    );
  };

export const withSocialLogin = <Props>(
  Component: React.ComponentType<{ onSocialLogin: onSocialLogin } & Props>,
) =>
  function withLoginHOC(props: Props) {
    return (
      <UserContext.Consumer>
        {({ onSocialLogin }: UserContextType) => (
          <Component {...props} onSocialLogin={onSocialLogin} />
        )}
      </UserContext.Consumer>
    );
  };

export const withUser = <Props>(
  Component: React.ComponentType<{ user: User } & Props>,
) =>
  function withUserHOC(props: Props) {
    return (
      <UserContext.Consumer>
        {({ user }: UserContextType) => <Component {...props} user={user} />}
      </UserContext.Consumer>
    );
  };

export const withLoginToken = <Props>(
  Component: React.ComponentType<{ loginToken: ?string } & Props>,
) =>
  function withLoginTokenHOC(props: Props) {
    return (
      <UserContext.Consumer>
        {({ loginToken }: UserContextType) => (
          <Component {...props} loginToken={loginToken} />
        )}
      </UserContext.Consumer>
    );
  };
