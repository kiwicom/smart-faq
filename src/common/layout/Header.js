// @flow

import { withRouter } from 'react-router-dom';
import * as React from 'react';
import idx from 'idx';
import css from 'styled-jsx/css';
import { TextLink } from '@kiwicom/orbit-components';
import Trans from '@kiwicom/nitro/lib/components/Text';

import { Desktop, Mobile } from '../Responsive';
import CloseButton from '../buttons/CloseButton';
import BackButton from '../buttons/BackButton';
import SignOutButton from './SignOutButton';
import { UserContext } from '../../context/User';
import { SelectedBooking } from '../../context/SelectedBooking';
import responsiveStyleHelperClasses from '../responsiveStyleHelperClasses';
import MobileBookingHeader from '../../MobileBookingHeader/MobileBookingHeader';
import type { UserContextType } from '../../context/User';
import SearchBar from '../../StaticFAQ/SearchBar';
import UserStatus from '../../helpers/UserStatus';

const style = css`
  .loggedOut {
    display: flex;
    justify-content: space-between;
    padding: 15px 122px 15px 40px;
  }
  .header {
    display: flex;
    align-items: center;
  }
  .HeaderFAQ {
    width: 100%;
    border-bottom: 1px solid #e8edf1;
    background-color: #ffffff;
  }
  @media only screen and (max-width: 500px) {
    .header.hide {
      opacity: 0;
      max-height: 0;
    }
  }
`;

const loggedInStyle = css`
  .helpHeader {
    font-size: 28px;
    font-weight: bold;
    color: #171b1e;
    pointer-events: none;
  }
  .loggedIn {
    display: flex;
    justify-content: space-between;
    flex: 1;
    padding: 15px 122px 15px 40px;
  }
  .links {
    display: flex;
    align-items: center;
  }
  a.open-icon {
    margin-left: 12px;
  }
  .staticFaqSearch {
    width: 350px;
    position: absolute;
    right: 260px;
    top: 10px;
  }
  @media only screen and (max-width: 900px) and (max-height: 480px) {
    .loggedIn {
      display: none;
    }
  }
  @media only screen and (max-width: 900px) {
    .loggedIn {
      padding: 15px;
    }
    .helpHeader {
      width: 100%;
      text-align: center;
      font-size: 1.2em;
    }
  }
`;

const loggedOutStyle = css`
  .helpHeader {
    position: absolute;
    left: 0;
    right: 0;
    text-align: center;
    font-size: 28px;
    font-weight: bold;
    color: #171b1e;
    pointer-events: none;
    margin: 0;
  }
  .loggedOut {
    display: flex;
    align-items: center;
    padding: 16px;
    height: 66px;
  }
  .signInOrBack {
    width: 43px;
    height: 20px;
  }
  .backButton {
    line-height: 2;
  }
  @media only screen and (max-width: 901px) {
    .helpHeader {
      position: absolute;
      left: 0;
      right: 0;
      text-align: center;
    }
  }
`;

type State = {
  isHeaderVisible: boolean,
  lastScroll: number,
  headerHeight: number,
};

type Props = {
  renderOnlyLoggedOut: boolean,
  history: {
    push: string => void,
  },
  match: {
    params: {
      categoryId?: string,
    },
  },
  location: {
    pathname: string,
  },
};

const renderLoggedIn = () => {
  return (
    <React.Fragment>
      <div className="loggedIn">
        <SelectedBooking.Consumer>
          {({ bookingPage }) => (
            <div className="helpHeader">
              {bookingPage === 'ALL_BOOKINGS' ? (
                <React.Fragment>
                  <Mobile>
                    <Trans t={__('smartfaq.mobile_header.title')} />
                  </Mobile>
                  <Desktop>
                    <Trans t={__('smartfaq.header.title')} />
                  </Desktop>
                </React.Fragment>
              ) : (
                <Trans t={__('smartfaq.header.title')} />
              )}
            </div>
          )}
        </SelectedBooking.Consumer>
        <Desktop>
          <div className="staticFaqSearch">
            <SearchBar />
          </div>
          <div className="links">
            <SignOutButton />
          </div>
        </Desktop>
        <style jsx>{loggedInStyle}</style>
        <style jsx>{responsiveStyleHelperClasses}</style>
      </div>
      <Mobile>
        <MobileBookingHeader />
      </Mobile>
    </React.Fragment>
  );
};

const renderLoggedOut = (
  hasCategory: string | null,
  isArticle: boolean,
  comesFromSearch: boolean,
  renderSignIn: boolean,
  push: string => void,
) => {
  const signIn = renderSignIn ? (
    <div className="desktopOnly" data-cy="sign-in-link">
      <TextLink
        external={false}
        onClick={() => push('/sign-in')}
        type="secondary"
      >
        <Trans t={__('smartfaq.header.sign_in')} />
      </TextLink>
    </div>
  ) : null;

  return (
    <div className="loggedOut">
      <div className="signInOrBack">
        {hasCategory || isArticle ? (
          <div className="backButton">
            <BackButton
              text={
                comesFromSearch
                  ? 'smartfaq.back_button.search'
                  : 'smartfaq.back_button.back'
              }
            />
          </div>
        ) : (
          signIn
        )}
      </div>
      <p className="helpHeader">
        <Trans t={__('smartfaq.header.title')} />
      </p>
      <style jsx>{responsiveStyleHelperClasses}</style>
      <style jsx>{loggedOutStyle}</style>
    </div>
  );
};

// prettier-ignore
type _SyntheticInputEvent = {
  target: { // eslint-disable-line react/no-unused-prop-types
    scrollTop: number,
    scrollLeft: number,
    scrollHeight: number,
    clientHeight: number,
  },
};

class Header extends React.Component<Props, State> {
  state = {
    isHeaderVisible: true,
    lastScroll: 0,
    headerHeight: 64, // eslint-disable-line react/no-unused-state
  };

  componentDidMount() {
    if (typeof window === 'undefined') {
      return;
    }

    window.addEventListener('scroll', this.handleScroll, true);
  }
  componentWillUnmount() {
    window.removeEventListener('scroll', this.handleScroll, true);
  }

  handleScroll = ({ target }: _SyntheticInputEvent) => {
    const tolerance = 6;
    const newHeaderHeight = idx(
      document.getElementById('SmartFAQ_header'),
      _ => _.offsetHeight,
    );
    const currentScroll = target.scrollTop;
    const Body = document.getElementById('SmartFAQ_Body');
    const distanceScrolled = Math.abs(this.state.lastScroll - currentScroll);

    if (currentScroll === undefined) {
      return;
    }

    const headerHeight = newHeaderHeight || this.state.headerHeight;

    const hide = () => {
      this.setState(prevState => {
        const headerHeight = newHeaderHeight || prevState.headerHeight;
        if (prevState.isHeaderVisible) {
          if (currentScroll < headerHeight) {
            target.scrollTop += headerHeight;
          }
          if (Body && screen.width < 900) {
            Body.style.marginTop = `${headerHeight}px`;
          }
        }
        return {
          isHeaderVisible: false,
          headerHeight: headerHeight,
          lastScroll: target.scrollTop,
        };
      });
    };

    const show = () => {
      this.setState(prevState => {
        if (Body) {
          Body.style.marginTop = '0';
        }
        return {
          isHeaderVisible: true,
          headerHeight: newHeaderHeight || prevState.headerHeight,
          lastScroll: currentScroll,
        };
      });
    };

    const isOnTop =
      currentScroll < headerHeight && currentScroll < this.state.lastScroll;

    if (distanceScrolled <= tolerance || distanceScrolled < 0) {
      if (isOnTop) show();
      this.setState({
        lastScroll: currentScroll,
      });
      return;
    }

    if (currentScroll > this.state.lastScroll) {
      hide();
      return;
    } else {
      show();
      return;
    }
  };

  render() {
    const hasCategory = idx(this.props.match, _ => _.params.categoryId) || null;
    const currentpath = this.props.location && this.props.location.pathname;
    const isArticle = currentpath.includes('article/');
    const comesFromSearch = currentpath.includes('faq/search/');
    const { renderOnlyLoggedOut } = this.props;

    return (
      <div
        id="SmartFAQ_header"
        className={!this.state.isHeaderVisible ? 'header hide' : 'header'}
      >
        <div className="HeaderFAQ">
          {!renderOnlyLoggedOut && (
            <UserStatus.LoggedIn>
              <Desktop>
                <CloseButton height="24" />
              </Desktop>
            </UserStatus.LoggedIn>
          )}
          <UserContext.Consumer>
            {({ simpleToken, loginToken }: UserContextType) => {
              const isLoggedIn = simpleToken || loginToken;
              const renderCloseButton = !isLoggedIn || renderOnlyLoggedOut;

              return (
                <React.Fragment>
                  {renderCloseButton && <CloseButton height="24" />}
                  {isLoggedIn && !renderOnlyLoggedOut
                    ? renderLoggedIn()
                    : renderLoggedOut(
                        hasCategory,
                        isArticle,
                        comesFromSearch,
                        !renderOnlyLoggedOut,
                        this.props.history.push,
                      )}
                </React.Fragment>
              );
            }}
          </UserContext.Consumer>
        </div>
        <style jsx>{style}</style>
      </div>
    );
  }
}

export const RawContentHeader = Header;

export default withRouter(Header);
