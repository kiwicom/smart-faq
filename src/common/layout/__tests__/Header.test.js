// @flow

import React from 'react';
import { render } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import { UserContext } from '../../../context/User';
import { RawContentHeader } from '../Header';

describe('ContentHeader', () => {
  const props = {
    renderOnlyLoggedOut: false,
    history: {
      entries: [],
      push: jest.fn(),
    },
    location: {
      pathname: '/content',
    },
    match: {
      params: {
        categoryId: '21',
      },
    },
  };

  const user = {
    user: {
      id: '1',
      email: 'joe.doe@example.com',
      firstname: 'Joe',
      lastname: 'Doe',
    },
    loginToken: 'AAAABBBBCCCC',
    simpleToken: null,
    onLogin: jest.fn(),
    onLogout: jest.fn(),
    onSocialLogin: jest.fn(),
  };

  it('should render loggedIn user header', () => {
    const wrapper = render(
      <UserContext.Provider value={user}>
        <Router>
          <RawContentHeader {...props} />
        </Router>
      </UserContext.Provider>,
    );
    expect(wrapper.find('.loggedIn')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render loggedOut user header', () => {
    const wrapper = render(
      <Router>
        <RawContentHeader {...props} />
      </Router>,
    );
    expect(wrapper.find('.backButton').text()).toEqual(
      'smartfaq.back_button.back',
    );
    expect(wrapper.find('.loggedOut')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without SignIn on Home page without booking', () => {
    const wrapper = render(
      <Router>
        <RawContentHeader
          {...props}
          renderOnlyLoggedOut
          match={{ params: {} }}
        />
      </Router>,
    );

    expect(wrapper.find('.signInOrBack').children()).toHaveLength(0);
  });
});
