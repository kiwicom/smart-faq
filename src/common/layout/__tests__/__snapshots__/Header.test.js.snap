// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader should render loggedIn user header 1`] = `
<div
  class="jsx-4290801200 header"
  id="SmartFAQ_header"
>
  <div
    class="jsx-4290801200 HeaderFAQ"
  >
    <div
      class="jsx-2880528156 jsx-2367066345 loggedIn"
    >
      <div
        class="jsx-2880528156 jsx-2367066345 helpHeader"
      >
        smartfaq.header.title
      </div>
    </div>
  </div>
</div>
`;

exports[`ContentHeader should render loggedOut user header 1`] = `
<div
  class="jsx-4290801200 header"
  id="SmartFAQ_header"
>
  <div
    class="jsx-4290801200 HeaderFAQ"
  >
    <div
      class="jsx-477917638 close-icon"
      role="button"
      tabindex="-1"
    >
      <svg
        fill="currentColor"
        height="24px"
        preserveAspectRatio="xMidYMid meet"
        style="vertical-align:middle;color:#46515e"
        viewBox="0 0 24 24"
        width="24px"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      class="jsx-2367066345 jsx-1604692328 loggedOut"
    >
      <div
        class="jsx-2367066345 jsx-1604692328 signInOrBack"
      >
        <div
          class="jsx-2367066345 jsx-1604692328 backButton"
        >
          <div
            class="jsx-1190988628 back"
            data-cy="back-button"
            role="button"
            tabindex="-1"
          >
            <div
              class="jsx-1190988628 chevron"
            >
              <svg
                fill="currentColor"
                height="16px"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align:middle;color:#8291a6"
                viewBox="0 0 24 24"
                width="16px"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </div>
            <span
              class="Text__StyledText-inVtPV iqjyyW"
            >
              smartfaq.back_button.back
            </span>
          </div>
        </div>
      </div>
      <p
        class="jsx-2367066345 jsx-1604692328 helpHeader"
      >
        smartfaq.header.title
      </p>
    </div>
  </div>
</div>
`;
